src/source.c

```diff
@@ -1858,6 +1858,43 @@ _dispatch_timers_get_delay(uint64_t nows[], struct dispatch_timer_s timer[],
 	return ridx;
 }
 
+
+#ifdef __linux__ 
+// in linux we map the _dispatch_kevent_qos_s  to struct kevent instead of struct kevent64.
+// we loose the kevent.ext[] members and the time out is based on relavite msec based time
+// vs. absolute nsec based time. For now we make the adjustments right here until the solution
+// to either extend libkqueue with a proper kevent64 API or removing kevent all together 
+// and move to a lower API (e.g. epoll or kernel_module. Also leeway is ignored.
+
+static void 
+kevent_set_delay(_dispatch_kevent_qos_s *ke, uint64_t delay,uint64_t leeway, uint64_t nows[]) 
+{
+	_dispatch_source_timer_now(nows, DISPATCH_TIMER_KIND_WALL); // called to return nows[]
+	
+	delay /= 1000000L;
+        if ((int64_t)(delay) <= 0) 
```

Comment:
tab/space issue.
we use 4-width tabs.

Our coding rules are (in vim speak):

```vimrc
set cinoptions=
set cinoptions+=Ls
set cinoptions+=:0,=1s
set cinoptions+=t0
set cinoptions+=+2s,(2s,u2s,W2s
set cinoptions+=j1,J1,m1
au FileType d,c,cpp,objc setlocal tw=80 ts=4 sw=4 noet
```

please fix the mixed indent here